tests/validation/tests/v3_api/test_kdm_changes.py

```diff
@@ -0,0 +1,134 @@
+    It then deploys a workload on each cluster,
+    checks service discovery - DNS resolution and
+    checks the ingress when enabled
+    Helper function - validate_custom_cluster_kdm() to create the AWS nodes
+    and add to the cluster
+    """
+    k8s_version_url = CATTLE_API_URL + K8S_VERSION_URL
+    plugin = PLUGIN
+    head = {'Authorization': 'Bearer ' + USER_TOKEN}
+    response = requests.get(k8s_version_url, verify=False, headers=head)
+    data = response.json()
+    print(data)
+    k8s_v = data["value"]
+    default_k8s_versions = k8s_v.split(",")
+    # default_k8s_versions = ["v1.17.2-rancher1-1"]
+    list_process = []
+    print("response: ", response)
+    print("k8s_v: ", k8s_v)
+    print("default_k8s_versions: ", default_k8s_versions)
+    for k8s_version in default_k8s_versions:
+        rke_config_new = deepcopy(rke_config)
+        rke_config_new["kubernetesVersion"] = k8s_version
+        node_count = NODE_COUNT_KDM_CLUSTER * len(plugin)
+        aws_nodes = \
+            AmazonWebServices().create_multiple_nodes(
+                node_count, random_test_name(HOST_NAME))
+        i = 0
+        for plug in plugin:
+            if plug == "calico" or plug == "canal" or plug == "weave":
+                rke_config_new["network"]["options"] = \
+                    {"flannel_backend_type": "vxlan"}
+            rke_config_new["network"] = {"type": "networkConfig",
+                                         "plugin": plug}
+            client = get_user_client()
+            cluster_name = random_name()
+            cluster = client.create_cluster(name=cluster_name,
+                                            driver="rancherKubernetesEngine",
+                                            rancherKubernetesEngineConfig=
+                                            rke_config_new)
+            p1 = multiprocessing.Process(target=validate_custom_cluster_kdm,
+                                         args=(cluster,
+                                               rke_config_new,
+                                               aws_nodes
+                                               [i:i+NODE_COUNT_KDM_CLUSTER]))
+            CLUSTER_LIST.append(cluster)
+            list_process.append(p1)
+            p1.start()
+            i = i + NODE_COUNT_KDM_CLUSTER
+    for process in list_process:
+        process.join()
+    env_details = "env.CATTLE_TEST_URL='" + CATTLE_TEST_URL + "'\n"
+    env_details += "env.ADMIN_TOKEN='" + ADMIN_TOKEN + "'\n"
+    env_details += "env.USER_TOKEN='" + USER_TOKEN + "'\n"
+    failed_cluster = {}
+    i = 0
+    for cluster in CLUSTER_LIST:
+        env_details += \
+            "env.CLUSTER_NAME_" + str(i) + "='" + cluster.name + "'\n"
+        i = i + 1
+    create_config_file(env_details)
+    print("env_details:", env_details)
+    for cluster in CLUSTER_LIST:
+        try:
+            validate_cluster(client, cluster, cluster.state, False, False)
```

Comment:
Can we also include the cluster name/id for the cluster that succeeds validation?